source/game_sa/CStats.cpp

```diff
@@ -162,165 +235,307 @@ bool CStats::CheckForThreshold(float* pValue, float range) {
-        stat == STAT_SAWN_OFF_SHOTGUN_SKILL ||
-        stat == STAT_COMBAT_SHOTGUN_SKILL ||
-        stat == STAT_MACHINE_PISTOL_SKILL || 
-        stat == STAT_SMG_SKILL ||
-        stat == STAT_AK_47_SKILL || 
-        stat == STAT_M4_SKILL || 
-        stat == STAT_RIFLE_SKILL ||
-        stat == STAT_APPEARANCE ||
-        stat == STAT_ARMOR || 
-        stat == STAT_ENERGY || 
-        stat == STAT_DRIVING_SKILL ||
-        stat == STAT_FLYING_SKILL ||
-        stat == STAT_LUNG_CAPACITY || 
-        stat == STAT_BIKE_SKILL || 
-        stat == STAT_LUCK || 
-        stat == STAT_HORSESHOES_COLLECTED ||
-        stat == STAT_TOTAL_HORSESHOES || 
-        stat == STAT_OYSTERS_COLLECTED ||
-        stat == STAT_TOTAL_OYSTERS || 
-        stat == STAT_CYCLING_SKILL;
-}
-
-// Converted from cdecl void CStats::ProcessReactionStatsOnDecrement(uchar stat) 0x559730
-void CStats::ProcessReactionStatsOnDecrement(unsigned char stat) { plugin::Call<0x559730, unsigned char>(stat); }
+    switch (stat) {
+    case STAT_GIRLFRIEND_RESPECT:
+    case STAT_CLOTHES_RESPECT:
+    case STAT_FITNESS_RESPECT:
+    case STAT_FAT:
+    case STAT_STAMINA:
+    case STAT_MUSCLE:
+    case STAT_MAX_HEALTH:
+    case STAT_SEX_APPEAL:
+    case STAT_PISTOL_SKILL:
+    case STAT_SILENCED_PISTOL_SKILL:
+    case STAT_DESERT_EAGLE_SKILL:
+    case STAT_SHOTGUN_SKILL:
+    case STAT_SAWN_OFF_SHOTGUN_SKILL:
+    case STAT_COMBAT_SHOTGUN_SKILL:
+    case STAT_MACHINE_PISTOL_SKILL:
+    case STAT_SMG_SKILL:
+    case STAT_AK_47_SKILL:
+    case STAT_M4_SKILL:
+    case STAT_RIFLE_SKILL:
+    case STAT_APPEARANCE:
+    case STAT_ARMOR:
+    case STAT_ENERGY:
+    case STAT_DRIVING_SKILL:
+    case STAT_FLYING_SKILL:
+    case STAT_LUNG_CAPACITY:
+    case STAT_BIKE_SKILL:
+    case STAT_LUCK:
+    case STAT_HORSESHOES_COLLECTED:
+    case STAT_TOTAL_HORSESHOES:
+    case STAT_OYSTERS_COLLECTED:
+    case STAT_TOTAL_OYSTERS:
+    case STAT_CYCLING_SKILL:
+        return true;
+    }
+    return false;
+}
 
 // Converted from cdecl void CStats::CheckForStatsMessage() 0x559760
-void CStats::CheckForStatsMessage() { plugin::Call<0x559760>(); }
+void CStats::CheckForStatsMessage() {
+    plugin::Call<0x559760>();
+}
 
 // Converted from cdecl void CStats::LoadStatUpdateConditions(void) 0x559860
-void CStats::LoadStatUpdateConditions() { plugin::Call<0x559860>(); }
+void CStats::LoadStatUpdateConditions() {
+    plugin::Call<0x559860>();
+}
 
-// Converted from cdecl void CStats::LoadActionReactionStats(void) 0x5599B0
-void CStats::LoadActionReactionStats() { plugin::Call<0x5599B0>(); }
+// 0x5599B0
+void CStats::LoadActionReactionStats() {
+    CFileMgr::SetDir(gta_empty_string);
 
-// Converted from cdecl int CStats::FindMaxNumberOfGroupMembers(void) 0x559A50
-int CStats::FindMaxNumberOfGroupMembers() { return plugin::CallAndReturn<int, 0x559A50>(); }
+    FILESTREAM fp = CFileMgr::OpenFile("DATA\\AR_STATS.DAT", "rb");
+    char statName[64] = {}; // unused
```

Comment:
```suggestion
    char statName[64];
```
This is modern C++. Avoid this please.

Reply:
What do you think about using `%*s` specifier on sscanf? This string var is not needed at all.

Reply:
It's actually not even C++.
It's C99.
[Source](https://en.cppreference.com/w/c/language/array_initialization).

Reply:
I see, alright then.